### PR TITLE
Backup: replace ConnectButton with ConnectScreen

### PR DESCRIPTION
**Pull request by samiff**
#### Changes proposed in this Pull Request:

* In the Backup plugin, update current `<ConnectButton>` component with `<ConnectScreen>`.

#### Jetpack product discussion

* Internal: pbuNQi-1w3-p2
* Internal, task: 1199658893173824-as-1200603127282613/f
> The ConnectScreen will display when not connected...Manage your connection will only appear when actually connected

#### Does this pull request change what data or activity we track or use?

* No.

#### Testing instructions:

* Disconnect and deactivate Jetpack plugin.
* After pulling down these changes, activate the Jetpack Backup plugin and navigate to `/wp-admin/admin.php?page=jetpack-backup`
* You should see the ConnectScreen component active (`jp-connect-screen`), click the `Set up Jetpack` button to connect site and user.

### CODE REVIEW

**Comment by samiff** [comment]
Hey @sergeymitr 👋 

I'm updating the Backup plugin with the newer `<ConnectScreen>` component. I think things are looking pretty good, but would appreciate if you could have a look. One issue I'm struggling with is adding the `assetBaseUrl` similar to how `connection-ui` has done. I'm ending up with a doubled-up image URL it seems.

File: projects/plugins/backup/src/js/components/Admin.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ const Admin = () => {
 	const renderLoadedState = () => {
 		// Loading state
 		if ( ! connectionLoaded ) {
-			return renderJetpackConnection();
+			return renderConnectScreen();
```
Comment (jwebbdev) [inline projects/plugins/backup/src/js/components/Admin.js:92]: We can avoid displaying the screen while loading the connection by wrapping this in a hidden div.
```
return (
  <div style={ { 'display': 'none' } }>
    { renderConnectScreen() }
  </div>
);
```
We can probably also just remove the call to `renderPromptForConnection` entirely and replace it with `renderConnectScreen` in the `! connectionStatus.isUserConnected || ! connectionStatus.isRegistered` state.

**Review by jwebbdev — COMMENTED** [review]
I'll adjust some styles in a later PR to make the jp-hero not wrap around it.  I left a comment on how we could hide it while the connection is loading, but if that functionality is going to exist in the ConnectScreen component itself before v1 it isn't strictly necessary.

File: projects/plugins/backup/src/js/components/Admin.js (hunk 1 above)
Comment (sergeymitr) [reply projects/plugins/backup/src/js/components/Admin.js:92]: That was an oversight on my part.
I temporarily fixed that in #20355, and will implement the more elegant solution a bit later.

**Comment by sergeymitr** [comment]
> One issue I'm struggling with is adding the `assetBaseUrl` similar to how `connection-ui` has done. I'm ending up with a doubled-up image URL it seems.

Hi @samiff.
The difference seems to be that `connection-ui` only provides the relative image path, while the Backup plugin uses the absolute URL.
I adjusted the component to make the `assetBaseUrl` optional, so it works fine now. Please clean up the rest of the `assetBaseUrl` code from the Backup plugin, as it's not needed.
And I'll consider removing the `assetBaseUrl` property from the component altogether.

**Comment by jwebbdev** [comment]
Tested and working.  I'll follow up with some styling once approved / merged.

**Review by leogermani — APPROVED** [review]
Tested and it works fine.

We might have to adjust the connection flow, as I ended up in the "My plans" page in wordpress.com which I don't think it's the ideal experience, but since we are looking at all the flows in a separate task, and as John already reviewed this on, I'm approving this for now.

**Pull request merged**